### ISSUE TITLE
Seed is not independent from KKP Master OIDC Provider

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubermatic#11257]
### What happened?
During the investigation of https://github.com/kubermatic/kubermatic/issues/11250 me and @lsviben found out that:
1. An open connection from seed to master is needed to the endpoint `https://kkp.example.com/dex`
2. If a cluster uses OIDC, it depends on the master where the central DEX instance is running. If Dex is down or Master is not reachable, the access to the clusters will fail with OIDC kubeconfig access.
3. The constant trying to reach not reachable OIDC endpoint from the user cluster API server could case `client throtteling` and performance impacts.  

### Expected behavior
Somehow the seed should be work independently of the master and potential host his own DEX cache or something else.
The `kubeconfig` to a user cluster, should work by only have a running seed.

### How to reproduce the issue?

Place a firewall between Master <> Seed and only allow to reach the seed kubernetes API endpoint.

### How is your environment configured?

- KKP version: `2.21.2`
- Shared or separate master/seed clusters?: yes

### Provide your KKP manifest here (if applicable)

Could be shared via PM or you look at the later linked repo

<details>

```yaml
# paste manifest here
```

</details>

### What cloud provider are you running on?
Proxmox satic VMs (later kubevirt)

### What operating system are you running in your user cluster?

Rock Linux 8.6

### Additional information

Error message in the user cluster API server
```
E1028 14:45:00.626926       1 authentication.go:63] "Unable to authenticate the request" err="[invalid bearer token, oidc: authenticator not initialized]"
E1028 14:45:01.342717       1 oidc.go:335] oidc authenticator: initializing plugin: Get "https://kkp.example.com/dex/.well-known/openid-configuration": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
E1028 14:45:01.720250       1 authentication.go:63] "Unable to authenticate the request" err="[invalid bearer token, oidc: authenticator not initialized]"
```

**Comment by toschneck** [comment]
connected issues:
https://github.com/kubermatic/kubermatic/issues/11316
https://github.com/kubermatic/dashboard/issues/5220

**Comment by csengerszabo** [comment]
From @pkprzekwas :

> Debugging this issue resulted in a following summary:
> 
> 1. If you install KKP without setting OIDCKubeCfgEndpoint and OpenIDAuthPlugin feature gates, Kubernetes dashboard will no work.
> 2. Enabling those feature gates by default is blocked because we don't want to create a dependency between user clusters and master's Dex instance.
> 
> Action points:
> 
> 1. Improve dashboard by blocking Kubernetes dashboard button in case feature flags are missing (we should add a tooltip to let users why the button is blocked).
> 2. Investigate and write a proposal for running a replica of Dex configuration on seed clusters.

**Comment by pkprzekwas** [comment]
This issue should be now possible to solve with having Seed level OIDC provider. The feature has been released in 2.22, but haven't been battle tested yet as setting up a test environment is a bit cumbersome. 

(https://docs.kubermatic.com/kubermatic/v2.22/tutorials-howtos/oidc-provider-configuration/)
